### ISSUE TITLE
Multi-platform/architecture docker image support

### ISSUE THREAD

**Issue by kawerewagaba** [medic/cht-core#8277]
**Is your feature request related to a problem? Please describe.**
Setting up a local environment to build and test CHT `4.x` (and `3.x`) applications. I get this warning when i run `docker compose up`: `cht-upgrade-service The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`. The containers seem to be running, but the app is broken as shown below:

<img width="1512" alt="Screenshot 2023-05-28 at 08 46 22" src="https://github.com/medic/cht-core/assets/24539185/0c1c15a2-b328-49e8-9089-98ec14cd5eea">

<img width="1512" alt="Screenshot 2023-05-28 at 08 48 31" src="https://github.com/medic/cht-core/assets/24539185/03eb1427-ebe9-4dd2-a6bb-70a467c7d296">

I also noticed a couple errors, not sure they're related to the platform issue:

<img width="744" alt="Screenshot 2023-05-28 at 08 48 57" src="https://github.com/medic/cht-core/assets/24539185/621ee9f4-f484-4321-9359-79e10a58c392">

**Describe the solution you'd like**
Images that support both `linux/amd64` and `linux/arm64/v8` platforms.

**Describe alternatives you've considered**
Locally rebuilding the images and running the containers - for development - will see how this turns out, but seems to be a viable option. I should also point out that I'm closely following [this](https://forum.communityhealthtoolkit.org/t/issues-with-local-install/1905/10) discussion on the forum.

**Additional context**
Chip: Apple M2 Pro
macOS: Ventura 13.0

Docker desktop screenshot
<img width="965" alt="Screenshot 2023-05-28 at 08 33 00" src="https://github.com/medic/cht-core/assets/24539185/a274deb8-a210-4fbb-839f-845a66bd8c90">

**Comment by Hareet** [comment]
@kawerewagaba Nice issue :)
Right now, GitHub Actions doesn't support arm64 runners: https://github.com/actions/runner/issues/805#issuecomment-1438149537

It's in their plan for q4 2023, so hopefully we can wait it out. 

https://github.com/actions/runner-images/issues/2187
https://github.com/actions/runner-images/issues/5631

We build the images in CI, so I think its possible that we self-host a GH Action runner as seen from this post: https://aws.amazon.com/blogs/compute/building-arm64-applications-on-aws-graviton2-using-the-aws-cdk-and-self-hosted-runners-for-github-actions/

But our repository would need to heed this warning from the link above:
```
GitHub recommends only utilizing self-hosted runners with private repositories. 
Allowing self-hosted runners on public repositories and allowing workflows on public forks 
introduces a significant security risk.
```

tl;dr: I don't know how soon this will happen. I'd love arm64 4.x images, but we'd want to replicate CI tests and ensure the artifact hasn't changed. I'm not of the biggest fan of building it locally and distributing the arm64 images in case any dependency changes but that is something we could consider.

**Comment by dianabarsan** [comment]
Thanks a lot for reporting @kawerewagaba 

While we wait for GH support, maybe we could consider a manual step in our release process to build these images for other common architectures. 

I'm not sure how feasible this is, but if we're going to get support out of the box, I'm tempted to not over-engineer the workaround. 
Appreciate any inputs @garethbowen @craig-landry @Reagan

**Comment by garethbowen** [comment]
Manual makes sense to me assuming it's only final releases. It should only be a few builds assuming GH sticks to their roadmap.

**Comment by dianabarsan** [comment]
Closing as fixed via https://github.com/medic/cht-upgrade-service/issues/36
